csharp/src/Ice/TcpEndpoint.cs

```diff
@@ -265,20 +265,21 @@ private protected TcpEndpoint(TcpEndpoint endpoint, string host, ushort port)
         private protected override IConnector CreateConnector(EndPoint addr, INetworkProxy? proxy) =>
             new TcpConnector(this, addr, proxy);
 
-        internal virtual ITransceiver CreateTransceiver(IConnector connector, EndPoint addr, INetworkProxy? proxy)
+        internal virtual ITransceiver CreateTransceiver(IConnector connector, EndPoint addr, INetworkProxy? proxy, bool secure)
         {
             ITransceiver transceiver = new TcpTransceiver(Communicator, connector, addr, proxy, SourceAddress);
-            if (IsSecure)
+            if (IsAlwaysSecure || secure)
```

Comment:
It would be nice if TcpTransceiver could be secure and non-secure, just like TcpEndpoint, TcpConnection etc.

Currently, TcpTransceiver is non-secure only and we wrap it with SslTransceiver to get a secure transceiver.

Reply:
Sure, though I think it's out of the scope of this PR.

Reply:
Yeah, the plan is to merge TcpTransceiver and SSLTransceiver in a later PR.